### PR TITLE
Import selected PF components only

### PR DESCRIPTION
**Pull request by martinpitt**
This does not currently make any difference for the dist/ size because of https://github.com/patternfly/patternfly-react/issues/10618. But we do that in all other projects.

### CODE REVIEW

File: src/files-breadcrumbs.tsx [hunk 1]
```diff
@@ -16,11 +16,15 @@
  * You should have received a copy of the GNU Lesser General Public License
  * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
  */
-import cockpit from "cockpit";
 import React from "react";
+
 import { CheckIcon, HddIcon, PencilAltIcon, TimesIcon } from "@patternfly/react-icons";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Flex, FlexItem } from "@patternfly/react-core/dist/esm/layouts/Flex";
+import { Page } from "@patternfly/react-core/dist/esm/components/Page";
```
[the remark this answers is not in the document]
Comment (jelly) [reply src/files-breadcrumbs.tsx]: This sounds correct :)

[the remark this answers is not in the document]
Comment (martinpitt) [reply src/files-breadcrumbs.tsx]: Fixed.

File: src/files-breadcrumbs.tsx [hunk 2]
```diff
@@ -16,11 +16,15 @@
...
 import React from "react";
+
 import { CheckIcon, HddIcon, PencilAltIcon, TimesIcon } from "@patternfly/react-icons";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Flex, FlexItem } from "@patternfly/react-core/dist/esm/layouts/Flex";
+import { PageBreadcrumb } from "@patternfly/react-core/dist/esm/components/Page";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
 
-import { Button, Flex, FlexItem, PageBreadcrumb, TextInput } from "@patternfly/react-core";
+import cockpit from "cockpit";
```
Comment (jelly) [inline src/files-breadcrumbs.tsx:27]: What about:

```
src/dialogs/mkdir.tsx:} from '@patternfly/react-core';
src/fileActions.jsx:} from "@patternfly/react-core";
```

Etc?

Comment (martinpitt) [reply src/files-breadcrumbs.tsx:27]: Right, grep failure, thanks! will fix.

**Review by allisonkarlitskaya — COMMENTED** [review]
So, (puts on "not-a-js-person" hat):

If this actually decreases the bundle size and/or is standard practice then we should do it.  But: this looks pretty suboptimal.  I did some reading over the weekend and it looked like one of the main benefits of ESM was supposed to be that it didn't have side-effects on import and that you can therefore tree-shake it even if you're using a barrel file...

Is there some other change we could make to improve this that wouldn't require reaching inside of the module?

The `package.json` of `node_modules/@patternfly/react-core` features: `"main": "dist/js/index.js",`.  That entire directory (`dist/js/`) looks pretty cjs-ish (again, not knowing too much about this stuff) but there's also a `"module"` entry which points to an esm version of Patternfly.  I used our fancy new `-m`etadata mode and uploaded the result to the visualizer and it seems that we're getting the `esm` version...

**Review by allisonkarlitskaya — DISMISSED** [review]
Okay, so here's some interesting links:

 - https://esbuild.github.io/api/#tree-shaking-and-side-effects
 - https://github.com/patternfly/patternfly-react/pull/983

Basically, Patternfly has:

```
  "sideEffects": [
    "*.css",
    "*.scss"
  ],
```

because it apparently doesn't have its house clean in terms of css producing side-effects:

> Since the react-core package is NOT entirely pure ( the .css files have side effects via adding styles to the dom), the sideEffect property in the core package.json should reflect that. 
https://github.com/patternfly/patternfly-react/pull/983#discussion_r237687192

Note: I'm not even sure if this (side-effect-free css) is strictly possible.  I kinda get the logic: you actually kinda have to be careful when you write selectors and use class names to make sure you don't conflict with other things, right?  And the bundler certainly can't rule that out...  We even create our class names for icon purposes dynamically in a plugin, and those could even theoretically be delivered from the server... But it would be an awful limitation of the ecosystem if it wasn't somehow possible...

That would explain why the analyzer shows that the majority of our bundle size is straight-up css, most of it from Patternfly.

At the same time: this seems like something that would affect every package using CSS, if you wanted to be super strict about it.  And it doesn't seem to be the default: Patternfly had to specifically list css files as exempted.  So maybe the logic should be that we trust ourselves to write CSS in a way that it doesn't produce side-effects outside of the code that was directly including it?

Those are all questions that are outside of the scope of this project, though, so tl;dr: for better or for worse, I think this change makes sense.

File: src/files-breadcrumbs.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinpitt) [reply src/files-breadcrumbs.tsx:27]: Yes, we try to do the same: standard web platform - external modules -- local modules.

**Comment by martinpitt** [comment]
Indeed tree shaking is conceptually/practically relatively easy for JS (you keep a tree of exported and used identifiers by module). But this is impossible for CSS, as there is no requirement that every CSS rule has a "local" specific selector (like `.pf-v5-c-button`). In theory, PF's masthead component could define a `button` selector and globally affect everything. And classes can be added/removed at runtime, so it's the halting problem again.

So we can't expect bundlers to do this for us.

In practice though, PF is "side effect free" enough that we can do imports by component. There might be exceptions (like the `Page` component, but we import that everywhere anyway), but it seems to work well enough.

Note that I forgot many imports (I only grepped for `import.*react-core`), I'll fix that up after the more pressing pilot matters.

File: src/files-breadcrumbs.tsx (hunk 2 above)
Comment (martinpitt) [reply src/files-breadcrumbs.tsx:27]: Done.

**Review by allisonkarlitskaya — APPROVED** [review]
Looks good, thanks!

**Pull request merged**